### DISCUSSION TITLE
Using arrays in shortcodes

### DISCUSSION THREAD

**Discussion by smolcodes** [11ty/eleventy#1605] [Q&A]
I'm using eleventy-img and want to know is it possible to use arrays in eleventy shortcodes? I should note I'm using liquid template.

Here's the post front matter (a blog post):
```
---
title: my blog title
date: some-date-date
imageURL: /src/images/someimage.jpg
imageAlt: coffee
---
```
Here's the post list layout snippet

```html
     {% for item in postListItems %}
<article>
          {% Image {{ item.data.imageURL}} , {{ item.data.imageAlt }} %}
        <time style="font-size: 12px; font-family: Calibri;" datetime="{{ item.data.date }}">
        🕒 {{ item.data.date | dateFilter }}</time>
          <h2 class"w-full" ><a class="text-yellow-500 text-4xl font-serif" href="{{ item.url }}">{{ item.data.title }}</a></h2>
          <p  class="text-gray-800 font-sans text-base mb-5">{{ item.data.description }}</p>
 </article>
        {% endfor %}
```
I know this isn't possible but how would I go about doing something like this?

**Comment by pdehaan** [comment]
I haven't used eleventy-img yet, but I think this should work if you're using the shortcode on the same page as the front-matter:

```liquid
{% Image imageUrl, imageAlt %}
```

I don't think I understand what you mean by arrays in shortcodes. Is this in a loop or pagination somewhere?

**Reply by smolcodes** [reply]
Thanks for the reply,!

The shortcode isn't on the same page as the front matter. The front matter is for the individual post. The shortcode is in the postlist page. I'll change the code above for clarification.
